test/puppetlabs/trapperkeeper/services/authorization/authorization_service_test.clj

```diff
@@ -0,0 +1,58 @@
+(ns puppetlabs.trapperkeeper.services.authorization.authorization-service-test
+  (:require
+    [clojure.test :refer :all]
+    [clojure.string :as str]
+    [puppetlabs.trapperkeeper.app :refer [get-service]]
+    [puppetlabs.trapperkeeper.services :refer [defservice]]
+    [puppetlabs.trapperkeeper.services.authorization.authorization-service :refer [authorization-service]]
+    [puppetlabs.trapperkeeper.testutils.bootstrap :refer [with-app-with-config]]
+    [puppetlabs.trapperkeeper.testutils.logging :refer [with-test-logging]]
+    [ring.util.response :refer [response]]
+    [schema.test :as schema-test]))
+
+(use-fixtures :once schema-test/validate-schemas)
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Utilities
+
+(defprotocol EchoReverseService
+  (echo-reverse [this msg]))
+
+; A simple service using authorization via ring middleware
+(defservice
+  echo-reverse-service EchoReverseService [[:AuthorizationService wrap-handler]]
+  (echo-reverse
+    [this msg]
+    (let [handler (fn [req] (response (str msg (str/reverse (:body req)))))]
+      (wrap-handler handler))))
+
+(def minimal-config
+  "Minimal config required to bootstrap AuthorizationService with
+  EchoReverseService as a dependant consumer.
+
+  An empty list representing no rules with an expected behavior of
+  default-deny originating in the authorization library itself."
+  {:authorization {:rules []}})
+
+(def base-request
```

Comment:
That's what I was looking for and couldn't find.  It always trips me up that response is part of the ring API but request is instead tucked under the mock namespace.